payments-core/src/main/java/com/stripe/android/payments/core/authentication/threeds2/Stripe3ds2TransactionViewModel.kt

```diff
@@ -35,20 +32,48 @@ import com.stripe.android.stripe3ds2.transaction.Transaction
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.withContext
+import javax.inject.Inject
 import kotlin.coroutines.CoroutineContext
 
 internal class Stripe3ds2TransactionViewModel(
     private val args: Stripe3ds2TransactionContract.Args,
-    private val stripeRepository: StripeRepository,
-    private val analyticsRequestExecutor: AnalyticsRequestExecutor,
-    private val analyticsRequestFactory: AnalyticsRequestFactory,
-    private val threeDs2Service: StripeThreeDs2Service,
-    private val messageVersionRegistry: MessageVersionRegistry,
-    private val challengeResultProcessor: Stripe3ds2ChallengeResultProcessor,
     private val initChallengeRepository: InitChallengeRepository,
-    private val workContext: CoroutineContext,
     private val savedStateHandle: SavedStateHandle
 ) : ViewModel() {
+
+    @Inject
+    lateinit var stripeRepository: StripeRepository
+
+    @Inject
+    lateinit var analyticsRequestExecutor: AnalyticsRequestExecutor
+
+    @Inject
+    lateinit var analyticsRequestFactory: AnalyticsRequestFactory
+
+    @Inject
+    lateinit var messageVersionRegistry: MessageVersionRegistry
+
+    @Inject
+    lateinit var threeDs2Service: StripeThreeDs2Service
+
+    @Inject
+    lateinit var challengeResultProcessor: Stripe3ds2ChallengeResultProcessor
+
+    @Inject
+    @IOContext
+    lateinit var workContext: CoroutineContext
+
+    init {
+        DefaultPaymentAuthenticatorRegistry
+            .retrieveInstance(args.paymentAuthenticatorRegistryId)?.let { registry ->
+                registry.authenticationComponent.inject(this)
+            } ?: run {
+            throw IllegalStateException(
```

Comment:
This will happen if no one is holding `DefaultPaymentAuthenticatorRegistry` - which is not possible because `Stripe3ds2TransactionActivity` is started from `Stripe3ds2Authentcator`, which lives in the `DefaultPaymentAuthenticatorRegistry`.